tests/wf/wf_with_call.py

```diff
@@ -45,28 +48,18 @@ def wf_call(N, ngens, nsamples, survival=0.0, mutation_rate=0.0, simplify_interv
 
         dead = [(random.random() > survival) for k in pop]
         # this is: offspring ID, lparent, rparent, breakpoint
-        new_inds = [(next(labels), random.choice(pop), random.choice(pop), 
+        new_inds = [(next(labels), random.choice(pop), random.choice(pop),
                      random_breakpoint(), random_mutations(mutation_rate))
                     for k in range(sum(dead))]
-        j=0
         if debug:
             print("Replacing", sum(dead), "individuals.")
-        for offspring, lparent, rparent, bp, muts in new_inds :
-            if debug:
-                print("--->", offspring, lparent, rparent, bp)
-            while not dead[j] :
-                j+=1
-            pop[j]=offspring
-            j+=1
-            if bp > 0.0 :
-                records(parent=lparent, time=t, population=0, child=offspring,
-                        left=0.0, right=bp)
-            if bp < 1.0 :
-                records(parent=rparent, time=t, population=0, child=offspring,
-                        left=bp, right=1.0)
-            for mut in muts:
-                records.add_mutation(position=mut, node=input_id, 
-                                     derived_state=1, ancestral_state=0)
+        js, parents, times, pops, offsprings, lefts, rights = \
```

Comment:
rather than zipping and yielding, you could sample these seven things as independent vectors and use some numpy logical indexing, like
```
js = np.where(not dead)
breakpoints = [random_breakpoint() for _ in js]
mas = np.random.choice(pop, replace=True, size=sum(js > 0.0))
pas = np.random.choice(pop, replace=True, size=sum(js < 1.0))
offsprings = [next(labels) for _ in js]
childs = offsprings[js > 0.0] + offsprings[js < 1.0]
parents = mas + pas
lefts = [0.0 for _ in mas] + [x for x in breakpoints if x < 1.0]
rights = [x for x in breakpoints if x > 0.0] + [1.0 for _ in pas]
```
... dunno if that is any simpler, though?

Reply:
for now what I have on fc1a2b5 works, so I will leave it and try to fix simplify